.github/workflows/release.yml

```
@@ -8,20 +8,18 @@ jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v4
     - name: Set up Python
-      uses: actions/setup-python@v4
+      uses: actions/setup-python@v5
       with:
-        python-version: 3.11
+        python-version: 3.12
     - name: Install dependencies
       run: |
-        python -m pip install --upgrade pip build
-        pip install twine wheel
+        python -m pip install --upgrade pip build twine
     - name: Build the package
       run: |
         python -m build
-    - name: Upload to PyPI
-      run: twine upload dist/*
-      env: 
-        TWINE_USERNAME: ${{ secrets.TWINE_USERNAME }}
-        TWINE_PASSWORD: ${{ secrets.TWINE_PASSWORD }}
+    - name: Publish to PyPI
+      uses: pypa/gh-action-pypi-publish@v1
+      with:
+        password: ${{ secrets.TWINE_API_KEY }}
```

Under the hood this might use Twine but naming this for PyPI might be clearer for those coming here in future?

```suggestion
        password: ${{ secrets.PYPI_API_KEY }}
```

---

That works — I had named this before switching to the PyPA action!